### ISSUE TITLE
Want to delete host but stay at REMOVING status

### ISSUE THREAD

**Issue by dreamOW** [rancher/rancher#7122]
**Rancher Version: 1.14

**Docker Version:1.12.1

**Environment Type: Cattle

**Steps to Reproduce: At first, I found one of my host is dissconnected, there were some containers run in this host before,  then I want to delete it

**Results: but I find that the host is stay at REMOVING, and have no change any more

**Expected: so where I can fix this problems, by the way, it there any way I can re-connect my host if without delete it alreay dissconnect

**Comment by wzrdtales** [comment]
Same for me, but rancher version 1.2.1

**Comment by wzrdtales** [comment]
```
2016-12-19 13:50:43,752 ERROR [:] [] [] [] [ecutorService-7] [.e.s.i.ProcessInstanceDispatcherImpl] Unknown exception running process [host.remove:175522] on [26], canceled by [State [requested] is not valid for process [storagepoolhostmap.remove:null] on resource [39]] 
2016-12-19 13:50:43,784 ERROR [:] [] [] [] [cutorService-14] [.e.s.i.ProcessInstanceDispatcherImpl] Unknown exception running process [host.remove:169961] on [14], canceled by [State [requested] is not valid for process [storagepoolhostmap.remove:null] on resource [25]] 
2016-12-19 13:50:43,784 ERROR [:] [] [] [] [ecutorService-6] [.e.s.i.ProcessInstanceDispatcherImpl] Unknown exception running process [host.remove:176749] on [27], canceled by [State [requested] is not valid for process [storagepoolhostmap.remove:null] on resource [41]] 
```

**Comment by wzrdtales** [comment]
Downgraded back to 1.2.0 and the error disappeared, seems to have been introduced in 1.2.1

**Comment by wzrdtales** [comment]
Ok I've setup a new instance today, completely fresh and can confirm, this bug seems to have been introduced in 1.2.1, and does only occur there and only for cloud environments like Azure.

**Comment by qrpike** [comment]
Having this issue as well on:
Rancher: version 1.2.1
Kernel: 4.8.6
Docker: 1.12.3

**Comment by ibuildthecloud** [comment]
Thanks, I'll see if I can address this in 1.2.2.  Some of the cleanup logic may have adversely impacted host delete.

**Comment by sra** [comment]
I am seeing this as well on 1.2.1. Hosts running on EC2

The hosts are added and removed by spotinst, if it matters.

I hopped onto one of the hosts in this state and got the end of the network manager log:

```
time="2016-12-21T13:58:56Z" level=error msg="Failed to evaluate network state for 078b36b920a23f4a210ff32ae5362f5469d18dfceff848756bd3f6d739dc2a28: netplugin failed but error parsing its diagnostic message \"\": unexpected end of JSON input"
time="2016-12-21T13:58:56Z" level=error msg="Error processing event &docker.APIEvents{Action:\"die\", Type:\"container\", Actor:docker.APIActor{ID:\"078b36b920a23f4a210ff32ae5362f5469d18dfceff848756bd3f6d739dc2a28\", Attributes:map[string]string{\"io.rancher.project_service.name\":\"staging1-apps/proxy\", \"io.rancher.service.requested.host.id\":\"198\", \"com.redacted.desc\":\"Application request router\", \"io.rancher.loadbalancer.target.app-chat\":\"/cgi-bin/WebObjects/ChatServer.woa=56795\", \"io.rancher.loadbalancer.target.app-mobile\":\"/cgi-bin/WebObjects/MobileServer.woa=56793\", \"io.rancher.cni.wait\":\"true\", \"io.rancher.container.create_agent\":\"true\", \"io.rancher.service.hash\":\"ea398cec4b913bc3de1c54cc00f1a90ef056fc45\", \"io.rancher.loadbalancer.target.app-router\":\"/cgi-bin/WebObjects/Router.woa=56794\", \"io.rancher.service.deployment.unit\":\"45f0a39b-7238-4eb5-b86a-5cdbe08940cf\", \"io.rancher.cni.network\":\"ipsec\", \"io.rancher.container.hostname_override\":\"container_name\", \"io.rancher.loadbalancer.target.app-mmvxml\":\"/cgi-bin/WebObjects/MMVXML.woa=56790\", \"io.rancher.container.agent_id\":\"1486\", \"io.rancher.container.ip\":\"10.42.216.99/16\", \"io.rancher.container.uuid\":\"d15c0b09-a2d0-49e1-ac8b-97e19c3fb0a9\", \"io.rancher.loadbalancer.target.app-api\":\"/cgi-bin/WebObjects/MMAPI.woa=56792\", \"io.rancher.scheduler.global\":\"true\", \"image\":\"rancher/lb-service-haproxy:v0.4.6\", \"io.rancher.container.name\":\"staging1-apps-proxy-2\", \"io.rancher.loadbalancer.target.app-redacted\":\"/cgi-bin/WebObjects/redacted.woa=56789\", \"io.rancher.loadbalancer.target.app-redacted-rest\":\"/cgi-bin/WebObjects/redacted.woa/wa/Rest=56789\", \"io.rancher.project.name\":\"staging1-apps\", \"io.rancher.stack.name\":\"staging1-apps\", \"exitCode\":\"1\", \"io.rancher.container.agent.role\":\"environmentAdmin\", \"name\":\"r-staging1-apps-proxy-2-d15c0b09\", \"io.rancher.service.launch.config\":\"io.rancher.service.primary.launch.config\", \"io.rancher.stack_service.name\":\"staging1-apps/proxy\"}}, Status:\"die\", ID:\"078b36b920a23f4a210ff32ae5362f5469d18dfceff848756bd3f6d739dc2a28\", From:\"rancher/lb-service-haproxy:v0.4.6\", Time:1482328736, TimeNano:1482328736136302250}. Error: netplugin failed but error parsing its diagnostic message \"\": unexpected end of JSON input"
time="2016-12-21T13:58:57Z" level=info msg="CNI down" cid=29bdb8905309ee6544c810bcc37c023b4a4d2d80493e1675a53debc79c3eae7b networkMode=ipsec
nsenter: cannot open /proc/10545/ns/ipc: No such file or directory
```

**Comment by nicholasdille** [comment]
I am seeing this issue on Rancher 1.2.1 with hosts (Ubuntu 16.04 LTS) deployed on AWS. In one environment, the issues occurred and in another one it did not.

**Comment by deniseschannon** [comment]
Does anyone have any reproducible steps so I can try to reproduce it?

**Comment by francoishill** [comment]
Hi @deniseschannon.

Steps are:

- Register a host/agent (in my case a GCE Instance)
- Wait for it to show in Rancher UI
- Delete the Instance (before removing from the UI)
- Wait for it to show as "Disconnected" in UI
- Try and Deactivate+Delete it in Rancher

Rancher configuration:

- "Fresh" Rancher 1.2.1 installation and its database
- Default Rancher Environment (cattle)
- Using rancher-nfs

**Comment by sangeethah** [comment]
@francoishill , Thank you. I am able to reproduce the issue with the steps you have provided.

**Comment by nicholasdille** [comment]
@sangeethah In my case, the issue happened after I did the following:
1. Add a new host
2. Deactivate old host
3. Wait for services to be restarted on another host
4. Remove old host
At this point the old host is stuck in state "removing". Unfortunately, this happened in one of three environments. All environments are managed by the same Rancher server.

**Comment by sangeethah** [comment]
Tested with v1.2.2-rc2:

Cattle environment with few  hosts( from GCE) and rancher-nfs deployed.
Delete the host from GCE (before removing from the cattle UI)
Wait for it to show as "Disconnected" in UI
Try and Deactivate+Delete it in Rancher.

Host gets removed successfully.

@nicholasdille , there was a fix relating to failure in storagepoolhostmap.remove that was done.Fix should be available in v1.2.2 release.

**Comment by nicholasdille** [comment]
@sangeethah Thanks! Is there a way to remove those hosts manually before the update to v1.2.2 or will v1.2.2 take care of this automagically?
